### ISSUE TITLE
Skipping TOC when no headings (or print toc?)

### ISSUE THREAD

**Issue by KariRudjord** [nlbdev/pipeline#71]

**Comment by KariRudjord** [comment]
Bert can help you if needed, Ammar.

**Comment by josteinaj** [comment]
I think this is how it works now already?

@KariRudjord @matskober can you confirm?

**Comment by matskober** [comment]
Not sure what to test here: TOC is generated regardless of a print toc. That's how Kari wants it, I believe.

**Comment by KariRudjord** [comment]
When I try 560475, a book without headings, the only output I get is Title page and Om boka. The rest is empty pages and look strange. Job 567 @josteinaj and @matskober

**Comment by josteinaj** [comment]
I just updated Pipeline 2 so the job results from your job is gone. I tried running your job again, and I got all the content in the book, not only title page and om boka. I don't know why the bodymatter disappeared for you (if I understand correctly), but maybe you can try again now and see if it still happens?

**Comment by matskober** [comment]
There is no TOC generated, neither at the first volume nor the following volumes
[results (6).zip](https://github.com/nlbdev/pipeline/files/1130830/results.6.zip)

**Comment by josteinaj** [comment]
Ok, so does TOC generation behave as we want it now (except for the layout)?

**Comment by matskober** [comment]
[results (8).zip](https://github.com/nlbdev/pipeline/files/1130943/results.8.zip)
@KariRudjord I am not sure what you have decided: If no print toc, still a braille toc? If so, @josteinaj , the toc generator if failing to produce a braille toc, when no print toc is present in dtbook.

**Comment by matskober** [comment]
<img width="353" alt="toc" src="https://user-images.githubusercontent.com/8014898/27959658-3d137690-6320-11e7-81ac-25c80c334d78.PNG">
[results (9).zip](https://github.com/nlbdev/pipeline/files/1131088/results.9.zip)
Two items get the same page reference in the toc; seems to be counting from 1 both on the section containing the preface and on the section containing chapter 1

**Comment by bertfrees** [comment]
Is this only in the TOC or also on the corresponding pages?

**Comment by matskober** [comment]
The preface have no page numbers at all - and a bigger left margin is seems.

**Comment by bertfrees** [comment]
I'll have a closer look.

**Comment by bertfrees** [comment]
The page number of the "Kapittel ein" is 1 because of this CSS:

```css
counter-set: page 1;
```

**Comment by matskober** [comment]
Yeah, I thought so. Tried to include frontmatter, but it did not seem to work

**Comment by matskober** [comment]
```scss
frontmatter, bodymatter, body:not([epub|type~='cover'])/*:not([epub|type~='frontmatter'])*/:not([epub|type~='backmatter']) {
        page: body;
        counter-set: page 1;
}
```

**Comment by matskober** [comment]
see job #662

**Comment by bertfrees** [comment]
This resets the page numbers both when the frontmatter starts *and* when the bodymatter starts. I'm not sure exactly what you want to achieve. Do you want to not reset the page number at all? In that case it is really simple: just remove the `counter-set` rule.

**Comment by KariRudjord** [comment]
There is no toc, yes, but the headline Innhold is still on the page. Looking odd. Should not be an empty page there at all.

And for the chapters without headings:
Text in chapter without `<h1>` should start on row 6. Could that be fixed?

**Comment by matskober** [comment]
Why this big font, Kari?

**Comment by bertfrees** [comment]
I changed it. Kari sometimes puts angle brackets in her comments but these are interpreted specially in Markdown. Enclose XML tags in backticks (&#x60;) to avoid that.

**Comment by matskober** [comment]
So it's not her new style of leadership, then? 🙂

**Comment by bertfrees** [comment]
No luckily not :) I was intimidated a bit too at first.

**Comment by KariRudjord** [comment]
Well, it don't seem like I know what I'm doing, but Bert is luckily there to save me out of it :-)

**Comment by josteinaj** [comment]
To summarize the discussion so far, the only thing that needs to be done before we call this issue fixed is to remove the TOC when there are no headlines?

**Comment by KariRudjord** [comment]
Yes!

**Comment by bertfrees** [comment]
https://github.com/nlbdev/pipeline/issues/151 was a duplicate of this issue. I'll have a look now.

**Comment by bertfrees** [comment]
Done

**Comment by KariRudjord** [comment]
Works.
